(Web) Create Series layout

linb1 opened the pull request:
## Basecamp Scope
[(Web) Create Series layout](https://3.basecamp.com/3926363/buckets/27088350/todos/6419891226)

## What was done?

1. Add `ContentSeriesSingle` to render `ContentSeriesContentItem`
2. Added a component to truncate description text if they are too long
NEW:
3. Removed margins from `MediaItem`
4. Added Share button back

~~Note: Doesn't have share button which was on `ContentSingle` since it was excluded in Figma~~

## How to test?

1. Click on a `ContentSeriesContentItem`
2. Should follow design laid out in Figma: https://www.figma.com/file/aiXp20H5yMdkOWyXOvNNqh/Web-Embeds?type=design&node-id=149-36531&mode=dev
5. Text should truncate in description if too long (can see in video below)

https://github.com/ApollosProject/apollos-embeds/assets/68402088/c7f41f1a-c77a-45b7-8549-b6ab54b27376

UPDATED VIDEO:


https://github.com/ApollosProject/apollos-embeds/assets/68402088/516fdac4-566c-4665-a61c-2bab654ebbd7

jordanwade reviewed (commented):
@linb1 this is looking nice! Just a couple things, mostly nitpicky.

1. Can you add a TODO for the share button in BC. I think we should have a consistent expectation across the entire app that allows the user to share any type of content including a series.

2. Can you see if you can adjust the grid component to line up with the featured card. Or better yet remove the gutters on the left and right of the grid.
![CleanShot 2023-09-06 at 16 34 48@2x](https://github.com/ApollosProject/apollos-embeds/assets/2528817/bdf1a1aa-e559-404a-b418-b0cb8b03b2a8)

<img width="115" alt="CleanShot 2023-09-06 at 16 28 48@2x" src="https://github.com/ApollosProject/apollos-embeds/assets/2528817/ec2df7c7-ec08-4d98-8b0e-e59a47fe4c3a">

3. On mobile sizes can we get rid of the truncation?
<img width="234" alt="CleanShot 2023-09-06 at 16 37 01@2x" src="https://github.com/ApollosProject/apollos-embeds/assets/2528817/9e1faed2-7030-4490-92cf-1de24cf65b5f">

If any of these suggestions require us to alter the design a bit, we have the freedom to do that. I expect the share button might cause issues and I'm not overly excited about the truncation. So, if we can adjust the image or something to not need truncation unless the description/summary is REALLY long, that would be cool.

If you want to sync up on any of this feel free to ping me.

linb1 commented:
@jordanwade Added the share button back in, didn't deviate to far from the original design. I removed the margin on the `MediaItems` which were causing the gutters. Might cause somewhere else using those `MediaItem` to be a bit off, but I can go check + fix them as they come up.

I removed truncation all together for mobile, it definitely doesn't make sense to be there for mobile. For desktop, mainly for small-medium screens, longer descriptions do disrupt the position of the image, so I just have the image centered no matter what, and had the cutoff for truncation be 10 lines instead of 4. This can be adjusted pretty easily, so I can always increase the cutoff for when the text needs to be truncated. 

Let me know if this needs more tweaking 

https://github.com/ApollosProject/apollos-embeds/assets/68402088/516fdac4-566c-4665-a61c-2bab654ebbd7

jordanwade commented:
@linb1 can you fix the conflict on this when you have time, then it is good to merge.
